packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -965,8 +1256,28 @@ class ScaffoldState extends State<Scaffold> with TickerProviderStateMixin {
   void initState() {
     super.initState();
     _geometryNotifier = new _ScaffoldGeometryNotifier(null, context);
+    _fabPositioner = widget.fabPositioner ?? _kDefaultFabPositioner;
+    _fabMotionAnimator = widget.fabMotionAnimator ?? FabMotionAnimator.scaling;
+    _previousFabPositioner = _fabPositioner;
+    _fabMoveController = new AnimationController(vsync: this, lowerBound: 0.0, upperBound: 1.0, value: 1.0, duration: _kFloatingActionButtonSegue * 2);
+    _fabMoveController.addStatusListener(_onFabMoveStatus);
+  }
+
+  @override
+  void didUpdateWidget(Scaffold oldWidget) {
+    // Update the fab animator, and then schedule the fab for repositioning.
+    if (widget.fabMotionAnimator != oldWidget.fabMotionAnimator) {
+      setState(() {
+        _fabMotionAnimator = widget.fabMotionAnimator ?? FabMotionAnimator.scaling;
```

Comment:
no need to call setState from inside didUpdateWidget

Reply:
Done